### PR TITLE
argc 1.14.0 (new formula)

### PR DESCRIPTION
**Pull request by sigoden**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/a/argc.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/argc", "--argc-create", "build"
```
Comment (SMillerDev) [inline Formula/a/argc.rb:15]:
```suggestion
    system bin/"argc", "--argc-create", "build"
```

File: Formula/a/argc.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system bin/"argc", "--argc-create", "build"
+    assert_predicate testpath/"Argcfile.sh", :exist?
+    cd testpath do
+      assert_match "build", shell_output("#{bin}/argc build")
+    end
```
Comment (chenrui333) [inline Formula/a/argc.rb:19]: by default, it should be already in the testpath :)

```suggestion
    assert_match "build", shell_output("#{bin}/argc build")
```

File: Formula/a/argc.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  desc "Easily create and use cli based on bash script"
+  homepage "https://github.com/sigoden/argc"
+  url "https://github.com/sigoden/argc/archive/refs/tags/v1.14.0.tar.gz"
+  sha256 "4e1b3b01ef3dd590b1201a84d63b275ab65ce145793a1d15b57cfaa2149d3791"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
```
Comment (chenrui333) [inline Formula/a/argc.rb:11]: can you also try to add the completion support? 

- https://github.com/sigoden/argc/pull/170

Comment (sigoden) [reply Formula/a/argc.rb:11]: The bash completion of argc requires bash version 5 or later, will this be an issue?

Comment (chenrui333) [reply Formula/a/argc.rb:11]: I think it would be an issue, but we will see.

Comment (sigoden) [reply Formula/a/argc.rb:11]: Typically, argc is registered for completions along with the scripts written based on argc, such as `argc --argc-completions zsh mycmd1 mycmd2 ...`. Or use with https://github.com/sigoden/argc-completions

So initially I didn't add the completion.
I'm not sure if it's necessary to generate completion only for argc.

Comment (chenrui333) [reply Formula/a/argc.rb:11]: yeah, if upstream supports the completion generation, then why not?

**Review by chenrui333 — APPROVED** [review]
LGTM, thanks!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**